File: src/main/java/jb/tester/MVScoreTester.java
package jb.tester;

import de.bwaldvogel.liblinear.Linear;
import de.bwaldvogel.liblinear.Model;
import jb.data.Problem;
import jb.data.ValidatingTestingTuple;

import java.util.List;

public class MVScoreTester {

    public double test(List<Model> clfs, ValidatingTestingTuple validatingTestingTuple) {
        double score = 0;
        Problem problem = validatingTestingTuple.getTestingProblem();
        for (int i = 0; i < problem.getY().length; i++) {
            double votingResult = 0;
            for (Model model : clfs) {
                votingResult += Linear.predict(model, problem.getObjectAsFeatures(i));
            }
            if (votingResult / clfs.size() > .5 ^ problem.getY()[i] == 1) {
                score++;
            }
        }
        score /= problem.getY().length;
        return score > .5 ? score : 1 - score;
    }
}

File: src/main/kotlin/jb/tester/MVScoreTester.kt
package jb.tester

import de.bwaldvogel.liblinear.Linear
import de.bwaldvogel.liblinear.Model
import jb.data.Problem
import jb.data.ValidatingTestingTuple

class MVScoreTester {

    fun test(clfs: List<Model>, validatingTestingTuple: ValidatingTestingTuple): Double {
        var score = 0.0
        val problem = validatingTestingTuple.testingProblem
        for (i in 0 until problem.y.size) {
            var votingResult = 0.0
            for (model in clfs) {
                votingResult += Linear.predict(model, problem.getObjectAsFeatures(i))
            }
            if ((votingResult / clfs.size > .5) xor (problem.y[i] == 1)) {
                score++
            }
        }
        score /= problem.y.size.toDouble()
        return if (score > .5) score else 1 - score
    }
}
